src/App.js

```diff
@@ -2,34 +2,59 @@ import React, { useState, useEffect } from 'react';
 import TodoList from './TodoList';
 import AddTodoForm from './AddTodoForm';
 
-const useSemiPersistentState = () => {
+function App() {
   const [todoList, setTodoList] = useState(
     JSON.parse(localStorage.getItem('savedTodoList')) || []
   );
-  
+
+  const [isLoading, setIsLoading] = useState(true);
+
   useEffect(() => {
-    localStorage.setItem('savedTodoList', JSON.stringify(todoList));
+    let isMounted = true;
+
+    const sideEffectPromise = new Promise((resolve, reject) => {
+      setTimeout(() => {
+        resolve({ data: { todoList: [...todoList] } });
+      }, 2000);
+    });
+
+    sideEffectPromise
+      .then((result) => {
+        if (isMounted) {
+          setTodoList(result.data.todoList);
+          setIsLoading(false);
+        }
+      })
+      .catch(() => {
+        if (isMounted) {
+          setIsLoading(false);
+        }
+      });
+
+    return () => {
+      isMounted = false;
+    };
```

Comment:
There's a lot going on here and I totally get that it's confusing. Based on this code and your submission comments I'm gonna assume you got some help from Chatgpt or StackOverflow, so I'll try and break it down a bit and follow that up with an example of what the instructions were really looking for here. 

```
let isMounted = true;
```
Here you're creating a variable to track whether or not the component is still mounted, which I'm not really sure the curriculum has covered yet. It's a good practice but is a bit advanced for where the class currently is.

```
const sideEffectPromise = new Promise((resolve, reject) => {
      setTimeout(() => {
        resolve({ data: { todoList: [...todoList] } });
      }, 2000);
    });
```
Here you're storing the `Promise` in a variable so that methods can be applied to the variable instead of chained onto the `Promise` constructor. Again, not technically wrong but not what the instructions were after. You should have also set the value of the `todoList` property to the initial/default value of the `todoList` state, and set the initial/default value of the `todoList` state to an empty array. 

```
sideEffectPromise
      .then((result) => {
        if (isMounted) {
          setTodoList(result.data.todoList);
          setIsLoading(false);
        }
      })
      .catch(() => {
        if (isMounted) {
          setIsLoading(false);
        }
      });
```
Here you're applying methods to the `sideEffectPromise` variable. If you weren't using a separate variable these could just chain right onto the last `)` on the `Promise` constructor and would work the same way. More stuff with `isMounted`, which is fine if you understand what it's checking but should probably be left out if you don't. The instructions also haven't asked you to use the `catch` method yet.

```
return () => {
  isMounted = false;
};
```
This is just a quick cleanup function at the end of the `useEffect` that sets `isMounted` to false when the component is unmounted. Again, fine if you understand what it's doing but is probably doing more harm than good if you don't. 

Finally, here's an example of what the directions were more specifically looking for. I know you mentioned that you plan to look over the assignment again, so hopefully with this visual the individual steps of the assignment will make more sense.
```
useEffect(() => {
    new Promise((resolve, reject) =>
      setTimeout(
        () =>
          resolve({
            data: {
              todoList: JSON.parse(localStorage.getItem("savedTodoList")) || [],
            },
          }),
        2000
      )
    ).then((result) => {
      setTodoList(result.data.todoList);
      setIsLoading(false);
    });
  }, []);
```